apps/editor.planx.uk/src/components/EditorNavMenu/components/EnvironmentSelect.tsx

```diff
@@ -0,0 +1,193 @@
+  padding: theme.spacing(1),
+  backgroundColor: theme.palette.background.dark,
+  color: theme.palette.common.white,
+}));
+
+const StyledCard = styled(Card)<{ selected?: boolean }>(() => ({
+  borderRadius: 5,
+}));
+
+const CardContent = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(1),
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "flex-start",
+}));
+
+const environments: Environment[] = [
+  {
+    name: "production",
+    description: "Used for editing content and publishing live flows",
+    url: "https://editor.planx.uk",
+  },
+  {
+    name: "staging",
+    description: "Used for testing new features and content",
+    url: "https://editor.planx.dev",
+  },
+  // Only show this on Pizzas
+  ...(import.meta.env.VITE_APP_ENV === "pizza"
+    ? [
+        {
+          name: "pizza",
+          description:
+            "Temporary environment used for testing new features and content",
+          url: window.location.href,
+        },
+      ]
+    : []),
+  // Only show this locally
+  ...(import.meta.env.VITE_APP_ENV === "development"
+    ? [
+        {
+          name: "development",
+          description: "Local development",
+          url: "http:/localhost:3000",
+        },
+      ]
+    : []),
+];
+
+export const EnvironmentSelect: React.FC = () => {
+  const [open, setOpen] = useState(false);
+  const currentEnv = import.meta.env.VITE_APP_ENV;
+  const { pathname } = useLocation();
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  return (
+    <>
+      <StyledButtonBase onClick={handleOpen} selected={false}>
+        {currentEnv}
+        <UnfoldMoreIcon sx={{ ml: 0.5 }} />
+      </StyledButtonBase>
+      <StyledDialog
+        open={open}
+        onClose={handleClose}
+        maxWidth="xs"
+        PaperProps={{
+          sx: {
+            position: "absolute",
+            top: 5,
+            left: 10,
+            m: 0,
+            width: "300px",
+            maxWidth: "300px",
+            minWidth: "unset",
+            borderTop: "none",
+            borderRadius: 3,
+          },
+        }}
+      >
+        <StyledDialogTitle>
+          <Box>
+            <Typography variant="body1" component="span" mr={1}>
+              Plan✕
+            </Typography>
+            <Typography variant="body1" component="span">
+              environments
+            </Typography>
+          </Box>
+          <IconButton
+            aria-label="close"
+            onClick={handleClose}
+            sx={{ color: "#ffffff", padding: 0 }}
+          >
+            <CloseIcon />
+          </IconButton>
+        </StyledDialogTitle>
+        <Stack p={1} bgcolor={(theme) => theme.palette.background.dark} gap={1}>
+          {environments.map((env) => (
+            <StyledCard key={env.name} selected={env.name === currentEnv}>
+              <CardActionArea
+                LinkComponent={"a"}
+                href={env.url + pathname}
```

Comment:
Unless already logged in to the destination env, this pathname won't get preserved due to this known bug - https://trello.com/c/047oaBJh/2962-fix-redirects-after-auth